### PR TITLE
kubevirt,performance: Bump presubmit and periodic performance lanes to kubernetes v1.31

### PR DESCRIPTION
**Pull request by brianmcarey**
**What this PR does / why we need it**:

With the introduction of the k8s-1.32 into kubevirtci[1] - the v1.29 provider will be removed soon.

Update the performance lanes to use the k8s-1.31 provider to keep them working

[1] https://github.com/kubevirt/kubevirtci/commit/4d4c8fe3fade746d3408715b06e9e9e741c3ab5b

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/cc @alaypatel07 @rthallisey @dhiller 

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note

```

### CODE REVIEW

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml [hunk 1]
```diff
@@ -1134,7 +1134,7 @@ periodics:
   # note: when the provider is updated, the `robots/cmd/perf-report-creator/graph.sh` and
   #       `robots/cmd/perf-report-creator/scrape.sh` needs to be updated as well, refer to this
   #       https://github.com/kubevirt/project-infra/pull/3032 for the needed changes
-  name: periodic-kubevirt-e2e-k8s-1.29-sig-performance
+  name: periodic-kubevirt-e2e-k8s-1.31-sig-performance
```
Comment (alaypatel07) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:1137]: @brianmcarey can you please also update the `robots/cmd/perf-report-creator/graph.sh` , ref: https://github.com/kubevirt/project-infra/pull/3032/

Comment (brianmcarey) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:1137]: Yes sure - sorry glanced over that in the comment.

**Review by alaypatel07 — APPROVED** [review]
/lgtm
/approve

**Pull request merged**